### PR TITLE
fix: Open data handler fix

### PR DESCRIPTION
**Pull request by Bingjiling**
Issue #, if available:

Description of changes:

Fix the following bugs:

Data validation bugs:
* Sha: Regex does not have * at the end, only matching one digit 
* Description: description for open data exceed the length requirement and have () <> and [] characters 

Study update bugs: 
* Update study is not updating DB with newly pulled open data, rather it’s using record just pulled from the DB by the find study method 
* The find study method has been updated as part of BYOB to add a status for a study, it’s not part of the update study schema and is failing the data validation 

Time out bug: 
* Timeout for Lambda was not set, it’s default to 6 seconds, the function can take up to 10 seconds sometime 

Permission bug: 
* Open data scrape handler role does not have permission for studyPermissions table, which is needed for update study 

WIP - Unit tests 

Checklist:

- [x] Have you successfully deployed to an AWS account with your changes?
- [x] Have you written new tests for your core changes, as applicable?
- [x] Have you successfully tested with your changes locally?
- [ ] Have you updated openapi.yaml if you made updates to API definition (including add, delete or update parameter and request data schema)?
- [ ] If you had to run manual tests, have you considered automating those tests by adding them to [end-to-end tests](../main/end-to-end-tests/README.md)?

AS review ticket id:

By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.

### CODE REVIEW

File: addons/addon-base-raas/packages/base-raas-services/lib/schema/create-study.json [hunk 1]
```diff
@@ -36,7 +34,8 @@
     "sha": {
       "type": "string",
       "maxLength": 64,
-      "pattern": "^([A-Fa-f0-9])$"
+      "pattern": "^([A-Fa-f0-9]*)$",
```
Comment (jn1119) [inline addons/addon-base-raas/packages/base-raas-services/lib/schema/create-study.json]: Should we use `{40}` to lock down its length? I

File: main/solution/backend/src/lambdas/open-data-scrape-handler/handler-impl.js [hunk 2]
```diff
@@ -183,24 +187,26 @@ const newHandler = async ({ studyService, log = consoleLogger } = {}) => {
...
+          const existingStudy = await studyService.find(userContext, study.id);
+          if (!existingStudy) {
+            await studyService.create(userContext, study);
+          } else {
+            await studyService.update(userContext, { rev: existingStudy.rev, ..._.omit(study, 'category') });
+          }
+          // Catch the err here so other open data update could continue
+        } catch (err) {
+          log.error(err);
+          log.error(study);
```
Comment (jn1119) [inline main/solution/backend/src/lambdas/open-data-scrape-handler/handler-impl.js]: Add some messaging here perhaps? `Study ${study} failed with error`...

File: main/solution/backend/src/lambdas/open-data-scrape-handler/handler-impl.js [hunk 3]
```diff
@@ -183,24 +187,26 @@ const newHandler = async ({ studyService, log = consoleLogger } = {}) => {
...
-          await studyService.update(userContext, studyToUpdate);
+        try {
+          const existingStudy = await studyService.find(userContext, study.id);
+          if (!existingStudy) {
+            await studyService.create(userContext, study);
+          } else {
+            await studyService.update(userContext, { rev: existingStudy.rev, ..._.omit(study, 'category') });
+          }
+          // Catch the err here so other open data update could continue
+        } catch (err) {
```
Comment (jn1119) [inline main/solution/backend/src/lambdas/open-data-scrape-handler/handler-impl.js]: Can we add a unit test here which checks that Promise.all succeeds and doesn't throw an Exception if one of the underlying calls fail.

File: addons/addon-base-raas/packages/base-raas-services/lib/schema/create-study.json (hunk 1 above)
Comment (jn1119) [inline addons/addon-base-raas/packages/base-raas-services/lib/schema/create-study.json]: Can we add a unit test for this?

File: addons/addon-base-raas/packages/base-raas-services/lib/study/__tests__/study-service.test.js [hunk 4]
```diff
@@ -17,6 +17,7 @@ const _ = require('lodash');
 const ServicesContainer = require('@aws-ee/base-services-container/lib/services-container');
 const JsonSchemaValidationService = require('@aws-ee/base-services/lib/json-schema-validation-service');
 const AwsService = require('@aws-ee/base-services/lib/aws/aws-service');
+const { getSystemRequestContext } = require('@aws-ee/base-services/lib/helpers/system-context');
```
Comment (SanketD92) [inline addons/addon-base-raas/packages/base-raas-services/lib/study/__tests__/study-service.test.js:20]: Nit: Is system context really necessary in a unit test? Can't we just mock the authorization logic and proceed with the test?

Comment (Bingjiling) [reply addons/addon-base-raas/packages/base-raas-services/lib/study/__tests__/study-service.test.js:20]: For open data create/update, study-service uses `isSystem` to check authorization, which was an imported function. It's not quite straightforward to mock compared to a in class `isAuthorized` function.

**Review by SanketD92 — APPROVED** [review]
LGTM

**Pull request merged**